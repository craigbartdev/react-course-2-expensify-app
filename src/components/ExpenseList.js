import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => ( //export for test
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id} {...expense}/>
                })
            )
        }
    </div>
);

const mapStateToProps = (state) => { //built in function that changes state in store
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);

