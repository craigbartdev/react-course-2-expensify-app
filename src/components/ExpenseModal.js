import React from 'react';
import Modal from 'react-modal';

export const ExpenseModal = (props) => (
    <Modal
        isOpen={props.modalVisible}
        onRequestClose={props.modalClose}
        contentLabel="Delete Expense"
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Are you sure that you would like to delete this expense?</h3>
        <button onClick={props.modalAction} className="modal__button remove">Yes</button>
        <button onClick={props.modalClose} className="modal__button cancel">Cancel</button>
    </Modal>
);

export default ExpenseModal;