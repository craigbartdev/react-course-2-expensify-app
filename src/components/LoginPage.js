import React from "react";
import { connect } from 'react-redux';
import { googleLogin, facebookLogin } from '../actions/auth';

export const LoginPage = ({ googleLogin, facebookLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control</p>
            <button className="button button--google" onClick={googleLogin}>Login with Google</button>
            <button className="button" onClick={facebookLogin}>Login with Facebook</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    googleLogin: () => dispatch(googleLogin()),
    facebookLogin: () => dispatch(facebookLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

