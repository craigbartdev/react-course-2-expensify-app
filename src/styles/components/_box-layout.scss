.box-layout {
    align-items: center; //centered with flexbox
    background: url('/images/bg.jpg'); //url from public directory perspective
    background-size: cover; //cover entire background
    display: flex; //for flexbox
    height: 100vh; //100% of viewport height
    justify-content: center; //centered with flexbox
    width: 100vw; //for viewport width
}

.box-layout__box {
    background: fade-out(white , .15); //fade white 15% still some of img
    border-radius: 3px; //rounded corners
    padding: $l-size $m-size;
    text-align: center; //centered text
    width: 25rem; //width of box
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1; //overwrites base line height
}