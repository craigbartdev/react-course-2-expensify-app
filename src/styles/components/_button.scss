.button {
    background: $light-blue;
    border: none;
    color: white;
    display: inline-block; //adds some top and bottom padding
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1; //less top and bottom padding
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888888;
}