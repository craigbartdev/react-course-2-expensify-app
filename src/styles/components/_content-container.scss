.content-container {
    margin: 0 auto; //evenly distributed on both sides
    padding: 0 $m-size; //some padding between sides
    max-width: 80rem;
}