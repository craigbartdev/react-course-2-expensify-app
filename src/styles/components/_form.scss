.form {
    display: flex;
    flex-direction: column; //everything top to bottom
    >* {
        margin-bottom: $m-size; //for all children of form
    }
}

.form__error  {
    margin: 0 0 $m-size 0;
    font-style: italic;
}