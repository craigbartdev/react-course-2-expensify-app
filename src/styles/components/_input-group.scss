.input-group {
    display: flex;
    flex-direction: column; //for small screens
    margin-bottom: $m-size; 

    @media (min-width: $desktop-breakpoint) {
        flex-direction: row; // for large screens
        margin-bottom: $l-size;
    }
}

.input-group__item {
    margin-bottom: $s-size;
    @media (min-width: $desktop-breakpoint) {
        margin: 0 $s-size 0 0; //get rid of margin-bottom with right margin
    }
}