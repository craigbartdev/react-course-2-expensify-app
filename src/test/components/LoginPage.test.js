import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "../../components/LoginPage";

test("should test LoginPage", () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should test the login button', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={ startLogin } />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});