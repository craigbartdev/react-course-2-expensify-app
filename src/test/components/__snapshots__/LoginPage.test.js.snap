// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test LoginPage 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
